src/Bundle/Config/ConfigResolver.php

```diff
@@ -62,16 +61,22 @@ public function getBundleConfigs($development): array
         return $this->order($bundles, $resolvedOrder);
     }
 
-    private function mergeConfig($otherConfig, ConfigInterface $config): ConfigInterface
+    private function mergeConfig(array $bundles, ConfigInterface $config): ?ConfigInterface
     {
-        if (!($otherConfig instanceof BundleConfig) || !($config instanceof BundleConfig)) {
-            throw new UnexpectedValueException('Mixing config classes is not supported.');
+        if (!isset($bundles[$config->getName()])) {
```

Comment:
This will now silently drop the "other" configuration instead of trying to merge it.
From my understanding, this is wrong as we will:
1. ignore defined information.
2. won't even tell the user that we are ignoring it.

How can I reproduce the original issue?

Reply:
Install any Contao 3 extension, e.g. Haste (see #30).

Reply:
Also, how does this "silently drop the other configuration"? If the `isset()` check fails, there is no other configuration, is there?

Reply:
Old way was:
- Check if there is already a config with that name
  - if so, try to merge or fail if impossible (the exception we now get)

New way is:
- Check if there is a config with that name
  - if so, check if both are bundles
    - if not, return null (therefore overwriting the already present config in https://github.com/contao/manager-plugin/pull/32/files#diff-be76487cdeb2e0d41287294fd3f8cbe7R50)

The information from the previous config has been lost without further notice then.

I'll have a look why this happens with Contao 3 extensions.

Reply:
Problem is in [contao/manager-bundle/src/ContaoManager/Plugin.php](https://github.com/contao/contao/blob/cdf31eff6ceffacbe53510b6a37d2cc252892b35/manager-bundle/src/ContaoManager/Plugin.php#L117):
```php
            foreach ($modules as $module) {
                $path = $module->getPathname();
                if (!file_exists(Path::join($module->getPathname(), '.skip'))) {
                    $iniConfigs[] = $parser->parse($module->getFilename(), 'ini');
                }
            }

            if (!empty($iniConfigs)) {
                $configs = array_merge($configs, ...$iniConfigs);
            }
```

We end up here with (in my case):
```
array (
  0 => 
  array (
    0 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'dcawizard',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'core',
        1 => 'haste',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
    1 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'core',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'calendar',
        1 => 'comments',
        2 => 'core',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
    2 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'haste',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'core',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
  ),
  1 => 
  array (
    0 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'haste',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'core',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
  ),
)
```
Note that `haste` is mentioned twice, despite only being installed "once".
My PR was based on the assumption that the modules only get loaded once (as they originate from `system/modules` and there may not be any two modules with the same name) which should remain true if not for a problem in [contao/manager-plugin/src/Bundle/Parser/IniParser.php#L39](https://github.com/contao/manager-plugin/blob/d377dcc93c0ec6a498948d2e9911fbd5ce267ca6/src/Bundle/Parser/IniParser.php#L39):
```php
    public function parse($resource, $type = null): array
    {
        $configs = [];
        $config = new ModuleConfig($resource);
        $configs[] = $config;

        $this->loaded[$resource] = true;
```
Here we mark the resource as loaded but FAIL to check initially if the resource HAS already been loaded in a previous call.
IMO this should be changed to:
```diff
     public function parse($resource, $type = null): array
     {
+        if (isset($this->loaded[$resource])) {
+            return [];
+        }
         $configs = [];
         $config = new ModuleConfig($resource);
         $configs[] = $config;
 
         $this->loaded[$resource] = true;
```
Which now (correctly yields:
```
array (
  0 => 
  array (
    0 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'dcawizard',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'core',
        1 => 'haste',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
    1 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'core',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'calendar',
        1 => 'comments',
        2 => 'core',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
    2 => 
    Contao\ManagerPlugin\Bundle\Config\ModuleConfig::__set_state(array(
       'name' => 'haste',
       'replace' => 
      array (
      ),
       'loadAfter' => 
      array (
        0 => 'core',
      ),
       'loadInProduction' => true,
       'loadInDevelopment' => true,
    )),
  ),
  1 => 
  array (
  ),
)
```
As the second call to load `haste` does not produce any more modules.